enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,442 @@
+- and the total change in the OCP subscription model between 3.x and 4.x:  you don't subscribe RHCOS nodes (in fact they
+do not have or ever will the entire set of subscription manager RPMs, and any as discussed in some of the alternatives
+below, at most we will only seed enough subscription manager artifacts in RHCOS so that things like turning auto mount
+on won't SEGFAULT).  In 3.x, the underlying linux was classic RHEL, and so hence the subscription manager was fully 
+good to go out of the box (and hence could be shared with pods running on that node).
+
+all those 3.x mechanisms are gone, and a series of steps from either the cluster admin and/or developer attempting
+to run OpenShift Builds leveraging entitled content must be performed after install to enable access to the entitled
+content during the build.
+
+The highlights of those steps from [the current 4.x doc](https://docs.openshift.com/container-platform/4.3/builds/running-entitled-builds.html):
+
+1) First, either
+ 
+- create an ImageStreamTag to the UBI image from `registry.redhat.io` in the `openshift` namespace, or 
+- reference the UBI image directly and include a pull secret to `registry.redhat.io` in your BuildConfig
+
+*NOTE*: in 4.5, image registry and build updates are occurring so that image streams and builds will have
+access to the `registry.redhat.io` credentials supplied in the install pull secret.  As such, the above steps
+in the manual process will no longer be required.
+
+2) Next, based on your access to the credentials:
+
+- With direct access to the entitlement pem files, create a generic secret from the files and reference the secret in
+your BuildConfig's input/source secret list
+- With Subscription Manager, and S2I build strategy, create secret(s) for the Subscription Manager configuration and 
+certs and reference the secret(s) in your BuildConfig's input/source secret list
+- With Subscription Manager, and Docker build strategy, you have to modify your Dockerfile and manually copy and 
+initialize as needed the information from the base image
+- With Satellite, and S2I build strategy, create a secret to reference the repository config file you created for your
+satellite instance and reference the secret in you BuildConfig's input/source secret list
+- With Satellite, and Docker build strategy, you have to modify your Dockerfile and manually copy and initialize as 
+needed the information from the base image as well as your repository config file for your satellite instance
+- And remember to set the skip layers optimization option (i.e. squash layers) with Docker Strategy builds
+
+*Another note on Satellite*: you can also subscribe nodes with Satellite (reference the "jump-host"), which results in 
+the pem files being placed in a well known location on the file system.  A user with sufficient privilege to the node 
+could obtain the pem files. 
```

Comment:
Can confirm this workflow, we tested this on a 4.3, subscribed the jump-host and used those pem files for cluster-wide entitlement with MCO as described here: https://docs.google.com/document/d/1-Il7O86vQBTKq2QNe47kQF-66n1NHgM9drEmtKWeELU/edit#heading=h.igkdrwoly0rc